include/fiction/layouts/coordinates.hpp

```diff
@@ -1017,6 +1054,119 @@ class coord_iterator
     CoordinateType coord;
 };
 
+/**
+ * Struct representing the aspect ratio of the cartesian layout.
+ *
+ * The `aspect_ratio` struct defines the starting and ending coordinates, effectively
+ * determining the size and position of the layout within a coordinate space.
+ */
+template <typename CoordinateType>
+struct aspect_ratio
```

Comment:
It could make sense to move the `area` and `volume` functions into `aspect_ratio`.